clang/test/CodeGen/fp-accuracy.c

```diff
@@ -0,0 +1,46 @@
+// RUN: %clang_cc1 -triple x86_64-unknown-unknown -ffp-accuracy-attr=high \
+// RUN: -Wno-implicit-function-declaration -emit-llvm -o - %s \
+// RUN: | FileCheck %s
+
+// RUN: %clang_cc1 -triple x86_64-unknown-unknown \
+// RUN: "-ffp-accuracy-attr=high:[sin,cosf] low:[tan] medium:[sincos]" \
+// RUN: -Wno-implicit-function-declaration -emit-llvm -o - %s \
+// RUN: | FileCheck --check-prefix=CHECK-FUNC %s
+
+// RUN: %clang_cc1 -triple spir64-unknown-unknown \
+// RUN: -D SPIR -ffp-accuracy-attr=sycl \
+// RUN: -Wno-implicit-function-declaration -emit-llvm -o - %s \
+// RUN: | FileCheck --check-prefix=CHECK-SPIR %s
+
+float a, b, c, d, e, f;
+
+void foo(void) {
+  // CHECK-LABEL: define {{.*}}void @foo()
+  a = cosf(b);
+  c = sin(a);
+  d = tan(c);
+  e = cos(d);
+#ifndef SPIR
+  d = sincos(e);
+#endif
+  // CHECK: call float @llvm.fpbuiltin.cos.f32(float {{.*}})        [[ATTR3:#[0-9]+]]
+  // CHECK: call double @llvm.fpbuiltin.sin.f64(double {{.*}})      [[ATTR3:#[0-9]+]]
+  // CHECK: call double @llvm.fpbuiltin.tan.f64(double {{.*}})      [[ATTR3:#[0-9]+]]
+  // CHECK: call double @llvm.fpbuiltin.cos.f64(double {{.*}})      [[ATTR3:#[0-9]+]]
+
+  // CHECK-FUNC: call float @llvm.fpbuiltin.cos.f32(float {{.*}})   [[ATTR3:#[0-9]+]]
+  // CHECK-FUNC: call double @llvm.fpbuiltin.sin.f64(double {{.*}}) [[ATTR3:#[0-9]+]]
+  // CHECK-FUNC: call double @llvm.fpbuiltin.tan.f64(double {{.*}}) [[ATTR4:#[0-9]+]]
+  // CHECK-FUNC: call double @llvm.fpbuiltin.cos.f64(double {{.*}})
```

Comment:
Why is the fpbuiltin intrinsic generated in this case? If it is generated, the test should do something to verify that no attribute is attached.